relay-cardinality/src/limiter.rs

```diff
@@ -199,7 +200,7 @@ impl<T: Limiter> CardinalityLimiter<T> {
 #[derive(Debug, Default)]
 struct DefaultReporter<'a> {
     exceeded_limits: HashSet<&'a CardinalityLimit>,
-    entries: HashSet<usize>,
+    entries: HashMap<usize, SmallVec<[String; 4]>>,
```

Comment:
We can Also hide the `SmallVec` behind a new-type `struct LimitsOrSomeBetterName<'a>(SmallVec<[&'a CardinalityLimit; 1]>)`

Makes changing the type later easier and nicer to pass around.